pkg/algorithm/cstorpoolselect/v1alpha1/select_test.go

```diff
@@ -157,3 +165,152 @@ func TestNewSelection(t *testing.T) {
+		"Test 8":  {[]string{}, []string{"uid 1", "uid 2", "uid 3"}, []string{"uid 1", "uid 2", "uid 3"}, "label3", false},
+		"Test 9":  {[]string{"uid 1", "uid 2", "uid 3"}, []string{"uid 1", "uid 5", "uid 3"}, []string{"uid 5"}, "label4", false},
+		"Test 10": {[]string{"uid 1", "uid 2", "uid 3", "uid 4"}, []string{"uid 1", "uid 2"}, []string{"uid 1", "uid 2"}, "label5", false},
+		"Test 11": {[]string{"uid 1", "uid 2", "uid 3", "uid 4"}, []string{"uid 1", "uid 2", "uid 3", "uid 5"}, []string{}, "label6", true},
+		"Test 12": {[]string{"uid 1", "uid 2", "uid 3", "uid 4"}, []string{"uid 1", "uid 2", "uid 3", "uid 5"}, []string{"uid 5"}, "label5", false},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			fakeList := &apis.CStorVolumeReplicaList{}
+			for _, uid := range test.CVRUid {
+				fakeList.Items = append(fakeList.Items,
+					apis.CStorVolumeReplica{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								string(replicaAntiAffinityLabel): test.labelSelector,
+								cstorPoolUIDLabelKey:             uid,
+							},
+						},
+					},
+				)
+			}
+
+			var fakeErr error
+			if test.expectError {
+				fakeErr = errors.New("Some fake error")
+			}
+
+			a := preferAntiAffinityLabel{
+				antiAffinityLabel: antiAffinityLabel{
+					labelSelector: test.labelSelector,
+					cvrList: func(name string, opts metav1.ListOptions) (*apis.CStorVolumeReplicaList, error) {
+						return fakeList, fakeErr
+					},
+				},
+			}
+
+			output, err := a.filter(test.poolUids)
+			if test.expectError && err == nil {
+				t.Fatalf("test %q failed: expected error not to be nil", name)
+			} else if !test.expectError && err != nil {
+				t.Fatalf("test %q failed: expected error to be nil", name)
+			} else if len(test.expectedUids) != len(output) {
+				t.Fatalf("test %q failed: expected %v but got %v", name, test.expectedUids, output)
+			} else if len(output) != 0 && !reflect.DeepEqual(test.expectedUids, output) {
+				t.Fatalf("test %q failed: expected %v but got %v", name, test.expectedUids, output)
+			}
+		})
+	}
+}
+
+func TestIsPolicy(t *testing.T) {
+	tests := map[string]struct {
+		policies     []policy
+		expectpolicy policyName
+		isPresent    bool
+	}{
+		"Test 1": {[]policy{&antiAffinityLabel{}}, antiAffinityLabelPolicyName, true},
+		"Test 2": {[]policy{&antiAffinityLabel{}}, antiAffinityLabelPolicyName, true},
+		"Test 3": {[]policy{&preferAntiAffinityLabel{}}, antiAffinityLabelPolicyName, false},
+		"Test 4": {[]policy{&preferAntiAffinityLabel{}}, antiAffinityLabelPolicyName, false},
+		"Test 5": {[]policy{&preferAntiAffinityLabel{}, &preferAntiAffinityLabel{}}, antiAffinityLabelPolicyName, false},
+		"Test 6": {[]policy{&preferAntiAffinityLabel{}, &preferAntiAffinityLabel{}}, preferAntiAffinityLabelPolicyName, true},
+	}
+
+	for name, test := range tests {
+		s := &selection{}
+		for _, p := range test.policies {
```

Comment:
Do we need to iterate ? Can we not instantiate the selection with the policies itself?

Reply:
yes we can do it